src/backend/src/services/projects.services.ts

```diff
@@ -0,0 +1,394 @@
+import { Role, User } from '@prisma/client';
+import { isProject, Project, WbsNumber, wbsPipe } from 'shared';
+import projectQueryArgs from '../prisma-query-args/projects.query-args';
+import prisma from '../prisma/prisma';
+import projectTransformer from '../transformers/projects.transformer';
+import { validateChangeRequestAccepted } from '../utils/change-requests.utils';
+import { AccessDeniedException, HttpException, NotFoundException } from '../utils/errors.utils';
+import {
+  addDescriptionBullets,
+  createChangeJsonNonList,
+  createRulesChangesJson,
+  editDescriptionBullets,
+  getHighestProjectNumber,
+  getUserFullName
+} from '../utils/projects.utils';
+import { descBulletConverter, wbsNumOf } from '../utils/utils';
+import { createDescriptionBulletChangesJson } from '../utils/work-packages.utils';
+
+export default class ProjectsService {
+  /**
+   * Get all the projects in the database.
+   * @returns all the projects
+   */
+  static async getAllProjects(): Promise<Project[]> {
+    const projects = await prisma.project.findMany({ where: { wbsElement: { dateDeleted: null } }, ...projectQueryArgs });
+    return projects.map(projectTransformer);
+  }
+
+  /**
+   * Get a single project
+   * @param wbsNumber the wbsNumber of the project to get
+   * @returns the request project
+   * @throws if the wbsNumber is invalid, the project is not found, or the project is deleted
+   */
+  static async getSingleProject(wbsNumber: WbsNumber): Promise<Project> {
+    if (!isProject(wbsNumber)) throw new HttpException(400, `${wbsPipe(wbsNumber)} is not a valid project WBS #!`);
```

Comment:
I think the original number was 404 and not 400, but I could be wrong though.

Reply:
it was but 404 is inaccurate becuase it's not a not found, it's a user error of sending something that isn't a valid project